### PR TITLE
CA-255021,XSO-672: When existing network is deact…

### PR DESCRIPTION
**Pull request by minglumlu**
…ivated and new network is added, 'Networking' tab of the VM shows incorrect information.

Signed-off-by: minglumlu <ming.lu@citrix.com>

### CODE REVIEW

**Comment by robertbreker** [comment]
Thanks for this!
A couple of asks from me:
- Could you please add details to the commit message on how this commit fixes the issue?
- Please run "gofmt -s -w ." to fix the syntax formating and make TravisCI happy
- Please format the commit as "CA-255021,XSO-672: When existing...." - this should allow bugtrackers to link to the ticket

**Comment by minglumlu** [comment]
This change is to make the guest agent to write the IP addresses of guests to xenstore with a new protocol described in http://xenbits.xen.org/gitweb/?p=xen.git;a=blob;f=docs/misc/xenstore-paths.markdown;hb=HEAD#l491
Example of old protocol:
> /local/domain/19/attr/eth0/ip = "10.71.78.243"
> /local/domain/19/attr/eth0/ipv6/0/addr = "fe80:0000:0000:0000:a872:13ff:feec:4a11"
> /local/domain/19/attr/eth0/ipv6/1/addr = "de80:0000:0000:0000:a872:13ff:feec:4a11"

Example of new protocol:
> /local/domain/19/attr/vif/0/ipv4/0 = "10.71.78.243"
> /local/domain/19/attr/vif/0/ipv6/0 = "fe80:0000:0000:0000:a872:13ff:feec:4a11"
> /local/domain/19/attr/vif/0/ipv6/1 = "de80:0000:0000:0000:a872:13ff:feec:4a11"

**Comment by minglumlu** [comment]
@robertbreker 
Thanks for your nice comments :)
I was confused by the build failure.

Ming

File: guestmetric/guestmetric_linux.go [hunk 1]
```diff
@@ -75,74 +75,90 @@ func (c *Collector) CollectMemory() (GuestMetric, error) {
...
+		}
+		paths = append(paths, prefixPaths...)
 	}
 
 	for _, path := range paths {
 		iface := filepath.Base(path)
-		if addrs, err := EnumNetworkAddresses(iface); err == nil {
+		vifId, err := getVifId(iface)
+		if err != nil {
+			continue
```
Comment (cheng-z) [inline guestmetric/guestmetric_linux.go]: Seems the error message been hide. Can we print the error message to log?

**Comment by cheng-z** [comment]
Looks good to me. Only thing I want is about the error log.
Thanks.

File: guestmetric/guestmetric_linux.go (hunk 1 above)
Comment (minglumlu) [inline guestmetric/guestmetric_linux.go]: Yes. Actually I did that in development with a minor hack. But there is no such logging mechanism in module in current implementation yet. So I'm afraid much more changes would be checked in than the changes so far for a logging mechanism in module.

I'm happy to do that. But before that, can you please let me know if you think it is necessary also.

Comment (cheng-z) [inline guestmetric/guestmetric_linux.go]: I think it is will be impotent for cases that VIF cannot support.
Can we estimate how difficult it is to add the log?
There are examples in this repo.

Comment (minglumlu) [inline guestmetric/guestmetric_linux.go]: It's not difficult. I could do that in one day, including testing.

**Comment by cheng-z** [comment]
Merge this as this is a routine which will call every 60 seconds. Too many log itself will cause problem.
So let's skip current log.

**Pull request merged**